content/automate/ManageIQ/Service/Provisioning/StateMachines/Methods.class/__methods__/dialog_parser.rb

```diff
@@ -46,6 +47,14 @@ def add_password_value(sequence, option_key, value, options_hash)
   prefixed_option_key = 'password::dialog_' + option_key
   add_hash_value(sequence, stripped_option_key.to_sym, value, options_hash)
   add_hash_value(sequence, prefixed_option_key.to_sym, value, options_hash)
+
+  if special_password?(option_key.to_sym)
+    add_hash_value(sequence, option_key.to_sym, value, options_hash)
+  end
+end
+
+def special_password?(option_key)
```

Comment:
I would suggest we avoid the generic term `special`.  Seem reasonable to replace `special` with `provision_dialog` for the method and constant to give the logic more context.